model/sharing/oauth.go

```diff
@@ -123,31 +122,34 @@ func (m *Member) CreateSharingRequest(inst *instance.Instance, s *Sharing, c *Cr
 
 // countFiles returns the number of files that should be uploaded on the
 // initial synchronisation.
-func (s *Sharing) countFiles(inst *instance.Instance) int {
-	count := 0
+func (s *Sharing) countFiles(inst *instance.Instance) int64 {
+	var count int64
+	fs := inst.VFS()
 	for _, rule := range s.Rules {
 		if rule.DocType != consts.Files || rule.Local || len(rule.Values) == 0 {
 			continue
 		}
 		if rule.Selector == "" || rule.Selector == "id" {
 			for _, fileID := range rule.Values {
-				_ = vfs.WalkByID(inst.VFS(), fileID, func(name string, dir *vfs.DirDoc, file *vfs.FileDoc, err error) error {
-					if err != nil {
-						return err
-					}
-					if file != nil {
-						count++
+				dir, file, err := fs.DirOrFileByID(fileID)
+				if err != nil {
+					continue
+				}
+				if dir != nil {
+					if _, nb, err := fs.DirSizeAndCount(dir); err == nil {
+						count += nb
 					}
-					return nil
-				})
+				} else {
+					count += file.ByteSize
```

Comment:
I believe this should be `count++` since we're counting files and not their size.